src/sdk/main/src/Transaction.cc

```diff
@@ -947,6 +947,12 @@ void Transaction<SdkRequestType>::updateSourceTransactionBody(const Client* clie
 
   // Add derived Transaction fields to mSourceTransactionBody.
   addToBody(mImpl->mSourceTransactionBody);
+
+  // Assign auto-renew account if the transaction is auto-renew assignable
+  if (client && client->getOperatorAccountId().has_value() && autoRenewAssignable(mImpl->mSourceTransactionBody))
```

Comment:
I'm not sure if this `if` check is correct, what if the user assigns the transaction ID manually to an account that isn't the operator? I think it should use the `mImpl->mTransactionId.mAccountId`